cookbook_download_test.go

```diff
@@ -151,6 +198,54 @@ func TestCookbooksDownloadTo(t *testing.T) {
 		assert.Nil(t, err)
 		assert.Equal(t, "log 'this is a resource'", string(recipeBytes))
 	}
+
+	// Capture the timestamps to ensure that on-redownload of unchanged cookook,
+	// they show no modification (using this as a proxy to determine whether
+	// the file has been re-downloaded).
+	defaultPathInfo, err := os.Stat(defaultPath)
+	assert.Nil(t, err)
+
+	metaDataInfo, err := os.Stat(metadataPath)
+	assert.Nil(t, err)
+
+	err = client.Cookbooks.DownloadTo("foo", "0.2.1", tempDir)
+	assert.Nil(t, err)
+
+	defaultPathNewInfo, err := os.Stat(defaultPath)
+	assert.Nil(t, err)
+
+	metaDataNewInfo, err := os.Stat(metadataPath)
+	assert.Nil(t, err)
+
+	err = client.Cookbooks.DownloadTo("foo", "0.2.1", tempDir)
+	assert.Nil(t, err)
+
+	// If the file was not re-downloaded, we would expect the timestamp
+	// to remain unchanged.
+	assert.Equal(t, defaultPathInfo.ModTime(), defaultPathNewInfo.ModTime())
+	assert.Equal(t, metaDataInfo.ModTime(), metaDataNewInfo.ModTime())
+
+	err = os.Truncate(metadataPath, 1)
+	assert.Nil(t, err)
+
+	err = os.Chtimes(metadataPath, metaDataInfo.ModTime(), metaDataInfo.ModTime())
```

Comment:
not sure if this function is cross-platform but, I guess we would just test on Unix. 🤔